lua/lean/lsp.lua

```diff
@@ -33,4 +40,18 @@ function M.handlers.plain_goal_handler (_, method, result, _, _, config)
   return vim.lsp.util.open_floating_preview(markdown_lines, "markdown", config)
 end
 
+function M.handlers.plain_term_goal_handler (_, method, result, _, _, config)
+  config = config or {}
+  config.focus_id = method
+  if not (result and result.goal) then
+    return
+  end
+  local markdown_lines = vim.lsp.util.convert_input_to_markdown_lines(result.goal)
+  markdown_lines = vim.lsp.util.trim_empty_lines(markdown_lines)
+  if vim.tbl_isempty(markdown_lines) then
+    return
+  end
+  return vim.lsp.util.open_floating_preview(markdown_lines, "leaninfo", config)
```

Comment:
Why is the goal converted to markdown if the filetype is leaninfo?

Reply:
That's me bad-copypasting from the above. Will fix.